src/script/sigcache.cpp

```diff
@@ -109,17 +127,66 @@ void InitSignatureCache()
         (nElems * sizeof(uint256)) >> 20, nMaxCacheSize >> 20, nElems);
 }
 
+template <typename F>
+bool RunMemoizedCheck(const std::vector<uint8_t> &vchSig, const CPubKey &pubkey,
+                      const uint256 &sighash, uint32_t flags, bool storeOrErase,
+                      const F &fun) {
+    uint256 entry;
+    signatureCache.ComputeEntry(entry, vchSig, pubkey, sighash, flags);
+    if (signatureCache.Get(entry, !storeOrErase)) {
+        return true;
+    }
+
+    if (!fun()) return false;
+    
+    if (storeOrErase)
+    {
+        signatureCache.Set(entry);
+    }
+    return true;
+}
+
+bool CachingTransactionSignatureChecker::IsCached(const std::vector<uint8_t> &vchSig, const CPubKey &pubkey,
+                                                  const uint256 &sighash) const
+{
+    return RunMemoizedCheck(vchSig, pubkey, sighash, nFlags, true, [] { return false; });
+}
+
+
+#if 0 // TODO
 bool CachingTransactionSignatureChecker::VerifySignature(const std::vector<unsigned char> &vchSig,
```

Comment:
do we still need to keep this #if 0 section

Reply:
I don't love the technique currently undefed that deploys templates and anonymous functions in a 30 line function, whose purpose is seemingly to save 4 or so repeated lines of code.  I am planning to analyze what's exactly going on here more closely and probably break it back into 2 functions.

I think we should move forward and make these cleanups in the final PR that turns on Schnorr at the fork time.